### ISSUE TITLE
Get rust compilation working (with native linker)

### ISSUE THREAD

**Issue by mhart** [lambci/lambci#9]
This almost certainly depends on #8 or at least a subset of that work.

Basically, rust will install fine and a bunch of commands work out of the box (eg, `cargo build` seems to work for a number of projects):

```bash
export CARGO_HOME=$HOME/.cargo
export MULTIRUST_HOME=$HOME/.multirust
export RUSTUP_HOME=$HOME/.multirust/rustup

curl https://sh.rustup.rs -sSf | sh -s -- -y

export PATH=$HOME/.cargo/bin:$PATH
rustc --version
cargo --version
```

But when it comes to the linking stage (often happens in `cargo test`), then it will fail because there's no native linker (ie, no gcc/ld).

It would be great to know if there's a minimal gcc/ld setup that will work here – I know that the Windows version of rust bundles gcc with it, so it definitely should be possible.

**Comment by mhart** [comment]
Thanks to #8 – Rust compilation now works! Example build:

https://lambci-public-buildresults-e3xwlufrwb3i.s3.amazonaws.com/gh/mhart/test-ci-project/builds/87/24a2a3358efdc78aa41c5530e36b3e74.html

BUT, it's not all roses. Unfortunately the Rust installation itself is huge and leaves little room for Cargo deps, so many projects won't build.

I'll open a new issue to potentially find some solutions for this.

**Comment by chris-morgan** [comment]
See #24 about shrinking rustc.
